server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java

```diff
@@ -158,6 +158,33 @@ private static boolean verifyIsAggregationDataTypeMatched(
     }
   }
 
+  public static TSDataType getScalarFunctionDataType(String funcName, TSDataType dataType) {
+    if (funcName == null) {
+      throw new IllegalArgumentException("ScalarFunction Name must not be null");
+    }
+    if (!verifyIsScalarFunctionDataTypeMatched(funcName, dataType)) {
+      throw new SemanticException(
+          "Scalar functions [DIFF] only support numeric data types [INT32, INT64, FLOAT, DOUBLE]");
```

Comment:
This error messgae should be more generic. `verifyIsScalarFunctionDataTypeMatched` is used to check whether data type is matched, not only numeric data types and [DIFF] shuold be replaced with `funcName`